shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -927,6 +929,92 @@ public void autofill_testLifeCycle() {
 
+  @Test
+  public void autofill_testAutofillUpdatesTheFramework() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+      return;
+    }
+
+    TestAfm testAfm =
+        Shadow.extract(RuntimeEnvironment.application.getSystemService(AutofillManager.class));
+    FlutterView testView = new FlutterView(RuntimeEnvironment.application);
+    TextInputChannel textInputChannel = spy(new TextInputChannel(mock(DartExecutor.class)));
+    TextInputPlugin textInputPlugin =
+        new TextInputPlugin(testView, textInputChannel, mock(PlatformViewsController.class));
+
+    // Set up an autofill scenario with 2 fields.
+    final TextInputChannel.Configuration.Autofill autofill1 =
+        new TextInputChannel.Configuration.Autofill(
+            "1",
+            new String[] {"HINT1"},
+            new TextInputChannel.TextEditState("field 1", 0, 0, -1, -1));
+    final TextInputChannel.Configuration.Autofill autofill2 =
+        new TextInputChannel.Configuration.Autofill(
+            "2",
+            new String[] {"HINT2", "EXTRA"},
+            new TextInputChannel.TextEditState("field 2", 0, 0, -1, -1));
+
+    final TextInputChannel.Configuration config1 =
+        new TextInputChannel.Configuration(
+            false,
+            false,
+            true,
+            TextInputChannel.TextCapitalization.NONE,
+            null,
+            null,
+            null,
+            autofill1,
+            null);
+    final TextInputChannel.Configuration config2 =
+        new TextInputChannel.Configuration(
+            false,
+            false,
+            true,
+            TextInputChannel.TextCapitalization.NONE,
+            null,
+            null,
+            null,
+            autofill2,
+            null);
+
+    final TextInputChannel.Configuration autofillConfiguration =
+        new TextInputChannel.Configuration(
+            false,
+            false,
+            true,
+            TextInputChannel.TextCapitalization.NONE,
+            null,
+            null,
+            null,
+            autofill1,
+            new TextInputChannel.Configuration[] {config1, config2});
+
+    textInputPlugin.setTextInputClient(0, autofillConfiguration);
+    textInputPlugin.setTextInputEditingState(
+        testView, new TextInputChannel.TextEditState("", 0, 0, -1, -1));
+
+    final SparseArray<AutofillValue> autofillValues = new SparseArray();
+    autofillValues.append("1".hashCode(), AutofillValue.forText("focused field"));
+    autofillValues.append("2".hashCode(), AutofillValue.forText("unfocused field"));
+
+    // Autofill both fields.
+    textInputPlugin.autofill(autofillValues);
+
+    // Verify the Editable has been updated.
+    assertTrue(textInputPlugin.getEditable().toString().equals("focused field"));
+
+    // The autofill value of the focused field is sent via updateEditingState.
+    verify(textInputChannel, times(1))
+        .updateEditingState(anyInt(), eq("focused field"), eq(13), eq(13), eq(-1), eq(-1));
```

Comment:
Where do the 13s come from?

Reply:
`"focused field".length == 13`, the autofill implementation puts the caret at the end.